server/ocsp.go

```diff
@@ -195,12 +219,21 @@ func (oc *OCSPMonitor) getRemoteStatus() ([]byte, *ocsp.Response, error) {
 	oc.mu.Lock()
 	hc := oc.hc
 	oc.mu.Unlock()
+
 	var raw []byte
 	for _, u := range responders {
 		u = strings.TrimSuffix(u, "/")
-		raw, err = getRequestBytes(fmt.Sprintf("%s/%s", u, reqEnc), hc)
+		// Prefer to make POST requests first.
+		raw, err = postRequestBytes(u, reqDER, hc)
 		if err == nil {
 			break
+		} else {
+			// Fallback to use a GET request.
+			reqEnc := base64.StdEncoding.EncodeToString(reqDER)
+			raw, err = getRequestBytes(fmt.Sprintf("%s/%s", u, reqEnc), hc)
```

Comment:
If the POST is more reliable and the GET fails too, this loses our POST error response.  Any chance we can use `errors.Join()` to return both errors, instead of risking masking a solid POST error response with a poor HTTP proxy GET error response which doesn't tell us why the POST failed in the first place?

Reply:
good idea, I applied these fixes.